### PR TITLE
Created data model for form

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
This is my take on an interface for all the form data. I welcome your feedback.

### CODE REVIEW

File: packages/api/constants/form-structure.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+// TODO get values
+export enum MainSpecilaization {
+  SPEC_1 = 'Specialization 1',
+  SPEC_2 = 'Specialization 2',
+  SPEC_3 = 'Specialization 3',
+}
+
+// TODO get values
+export enum SubSpecialization {
```
Comment (fwpushan) [inline packages/api/constants/form-structure.ts:67]: Sub-specilaization, could be function/mapped value of MainSpecilaization  `f(mainSpec) -> SunSpec`

File: packages/api/constants/form-structure.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+export interface FormData {
+  // Page 1
+  first_name: string;
```
Comment (fwpushan) [inline packages/api/constants/form-structure.ts:3]: why snake style is used for property naming?

**Review by fwpushan — COMMENTED** [review]
Initial Draft looks good. But I've few recommendations 
1. The fields and the expected value range should be clearly mentioned in corresponding ticket. So, will recommend, go with agile approach add what we required for that story.
2.  If there are any cross package usage of these interfaces then declare it in either `common` package or `types` package.

**Review by arranfw — COMMENTED** [review]
Liking the direction so far, let's chat about this as a team today

**Comment by hitankar-freshworks** [comment]
I think we are going away from the NestJs conventions

**Comment by hitankar-freshworks** [comment]
I find nesting keep things readable. I like the suggest the following structure:
```
FormData {
 personalInformation: {
     firstName: string,
     .
     .
   }
  registrationDetails: {
    streamTypes: string;
   .
  .
  },
  deploymentDetails: {
  deployAnywhere: boolean;
 .
 .
 .
 }
}
```

File: packages/api/constants/form-structure.ts (hunk 1 above)
Comment (hitankar-freshworks) [reply packages/api/constants/form-structure.ts:67]: I am curious if it will  work with class-validator if we use generics like `Specialization<Type>`

**Pull request closed without merging**